src/libraries/System.Web.HttpUtility/src/System/Web/Util/HttpEncoder.cs

```diff
@@ -73,44 +73,37 @@ private static unsafe void HtmlAttributeEncodeInternal(string s, TextWriter outp
             }
             else
             {
-                int cch = s.Length - index;
-                fixed (char* str = s)
-                {
-                    char* pch = str;
-                    while (index-- > 0)
-                    {
-                        output.Write(*pch++);
-                    }
+                output.Write(s.AsSpan(0, index));
 
-                    while (cch-- > 0)
+                ReadOnlySpan<char> remaining = s.AsSpan(index);
+                for (int i = 0; i < remaining.Length; i++)
+                {
+                    char ch = remaining[i];
```

Comment:
FWIW, the above could also be simply:
```C#
foreach (char ch in s.AsSpan(index))
```
The compiler will rewrite that to be basically identical to what's above.